java/arcs/core/data/InformationFlowLabel.kt

```diff
@@ -26,7 +26,7 @@ sealed class InformationFlowLabel {
 
     /** Represents a boolean expression of [InformationFlowLabel] constants. */
     sealed class Predicate {
-        data class Label(val label: InformationFlowLabel) : Predicate() {
+        data class Label(val label: SemanticTag) : Predicate() {
```

Comment:
I was thinking that a base label could be more than a `SemanticTag`. For example, we might want to support claims   that data is from a `store`, `handle`, etc., That is why I had this as `InformationFlowLabel`. Granted that we don't have anything other than `SemanticTag` yet, but it might be good to keep this until we decide that other tags are no longer needed. 

If this is turning out to be incovenient, can we add a to `Predicate` as follows?

```
  constructor(label: SemanticTag) : this(InformationFlow.SemanticTag(semanticTag)) 
```

Reply:
Not sure why I did that... I think at one point I needed it, but I just reverted it and everything seems fine.